src/main/java/com/jvm_bloggers/admin_panel/mailing/MailingAddressUniquenessValidator.java

```diff
@@ -0,0 +1,46 @@
+package com.jvm_bloggers.admin_panel.mailing;
+
+import com.jvm_bloggers.core.mailing.domain.MailingAddressRepository;
+import org.apache.wicket.markup.html.form.Form;
+import org.apache.wicket.markup.html.form.FormComponent;
+import org.apache.wicket.markup.html.form.validation.AbstractFormValidator;
+import org.apache.wicket.spring.injection.annot.SpringBean;
+import org.apache.wicket.util.lang.Args;
+
+import java.util.Optional;
+
+public class MailingAddressUniquenessValidator extends AbstractFormValidator {
```

Comment:
Agree that IValidator would be much simpler but I can't use it here. The reason is that we need to have both id and address during validation. For new addresses we don't need id, but during edition we have to exclude address value that is currently edited from uniqueness check (otherwise edit and save without changing address will cause validation error). Or I am missing something ?